### ISSUE TITLE
How to add conditionals in current version?

### ISSUE THREAD

**Issue by eduo** [json-editor/json-editor#1322]
I have seen dependencies and oneofs, but they seem inefficient for what I want to do. I'm looking for a way I can implement small changes across the form based on initial questions.

- Example 1: In a form you choose whether you're a company or a person, depending on that in a separate section you're not asked for a last name but you're asked for an establishment date. If these were not in the same section oneOf would be OK, but they are in completely different places.

- Example 2: You select a state, from which a listing of districts is selectable which in turn validates the postal code (whose first two numbers are dicated by the district).

On this case the list of districts for the country has matching states, so it just needs filtering. They all have prefix numbers associated to the postal code can be validated against them. The first part in particular would require dozens of oneof entries, all 99% identical, and would require changing the form each there's changes in the table and hardcoding too many values.

**Comment by schmunk42** [comment]
We only have https://github.com/json-editor/json-editor#dependencies
Also take a look at https://github.com/json-editor/json-editor#custom-validation

If-else is json-schema v7 (currently not supported).

**Comment by eduo** [comment]
Thank you. I've been getting some results in my needs by restructuring things a lot.

Something I have't found a way of doing is showing a field only when a condition is met (in this case, the first letter of a text field, if it matches an array). 

I thought of using an intermediate field via a template but the issue is the same. I imagine it's just not possible without if-then-else.

**Comment by germanbisurgi** [comment]
You could [listen for changes](https://github.com/json-editor/json-editor#listen-for-changes) and implement the logic inside a custom watcher.

editor.watch('path.to.field',() => {
  // Do something
});

**Comment by eduo** [comment]
> You could [listen for changes](https://github.com/json-editor/json-editor#listen-for-changes) and implement the logic inside a custom watcher.

Yeah. I thought about it but wanted to try as much as possible using schema. I was able to get the same results using heavy restructuring and basic dependencies.

I could use more complex rules but at the moment this is enough. Until we have if…then…else I'll fall back to custom watchers if I can't solve using schema.
